editoast/src/views/scenario/macro_nodes.rs

```diff
@@ -0,0 +1,635 @@
+use crate::views::projects::ProjectError;
+use crate::views::projects::ProjectIdParam;
+use crate::views::scenario::ScenarioError;
+use crate::views::scenario::ScenarioIdParam;
+use crate::views::study::StudyError;
+use crate::views::study::StudyIdParam;
+use crate::views::AuthenticationExt;
+use crate::views::AuthorizationError;
+
+crate::routes! {
+    "/macro_nodes" => {list, create,},
+    "/macro_nodes/{node_id}" => {get, update, delete,},
+}
+
+editoast_common::schemas! {
+    MacroNodeForm,
+    MacroNodeResponse,
+    MacroNodeListResponse,
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "macro_node")]
+enum MacroNodeError {
+    #[error("Node '{node_id}', could not be found")]
+    #[editoast_error(status = 404)]
+    NotFound { node_id: i64 },
+}
+
+#[derive(IntoParams, Deserialize)]
+#[allow(unused)]
+struct MacroNodeIdParam {
+    node_id: i64,
+}
+
+#[derive(Debug, Deserialize, ToSchema)]
+struct MacroNodeForm {
+    position_x: f64,
+    position_y: f64,
+    full_name: Option<String>,
+    connection_time: i64,
+    labels: Tags,
+    trigram: Option<String>,
+    path_item_key: String,
+}
+
+impl MacroNodeForm {
+    pub fn into_macro_node_changeset(self, scenario_id: i64) -> Result<Changeset<MacroNode>> {
+        let macro_node = MacroNode::changeset()
+            .scenario_id(scenario_id)
+            .position_x(self.position_x)
+            .position_y(self.position_y)
+            .full_name(self.full_name)
+            .connection_time(self.connection_time)
+            .labels(self.labels)
+            .trigram(self.trigram)
+            .path_item_key(self.path_item_key);
+
+        Ok(macro_node)
+    }
+}
+
+#[derive(Debug, Serialize, ToSchema)]
+#[cfg_attr(test, derive(Deserialize, PartialEq))]
+struct MacroNodeResponse {
+    id: i64,
+    position_x: f64,
+    position_y: f64,
+    full_name: Option<String>,
+    connection_time: i64,
+    labels: Tags,
+    trigram: Option<String>,
+    path_item_key: String,
+}
+
+impl From<MacroNode> for MacroNodeResponse {
+    fn from(node: MacroNode) -> Self {
+        Self {
+            id: node.id,
+            position_x: node.position_x,
+            position_y: node.position_y,
+            full_name: node.full_name,
+            connection_time: node.connection_time,
+            labels: node.labels,
+            trigram: node.trigram,
+            path_item_key: node.path_item_key,
+        }
+    }
+}
+
+impl PartialEq<MacroNodeResponse> for MacroNode {
+    fn eq(&self, other: &MacroNodeResponse) -> bool {
+        self.id == other.id
+            && self.position_x == other.position_x
+            && self.position_y == other.position_y
+            && self.full_name == other.full_name
+            && self.connection_time == other.connection_time
+            && self.labels == other.labels
+            && self.trigram == other.trigram
+            && self.path_item_key == other.path_item_key
+    }
+}
```

Comment:
You can move that in the `mod tests` below since it's not used in `cfg(not(test))`